services/order.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime.date = None
+) -> Order:
+    with transaction.atomic():
+        order = Order.objects.create(
+            user=get_user_model().objects.get(username=username)
+        )
+
+        if date:
+            order.created_at = date
+            order.save()
+
+        for ticket in tickets:
+            row, seat, movie_session_id = ticket.values()
+            Ticket.objects.create(
+                order=order,
+                row=row,
+                seat=seat,
+                movie_session_id=movie_session_id
+            )
```

Comment:
You can also try to use bulk_create here to not overload the db. Not necessary, just my opinion.

Reply:
Важливо враховувати деякі обмеження bulk_create(). Зокрема, при використанні цього методу не викликаються методи save()

Reply:
So in this example probably we can't use bulk_create because we have validation which connected to save(). And I had error in test like: ticket with this movie_session already presented

Reply:
Very strange... Maybe you did some mistake cause it should work in both ways